Subnet broadcasts

MabezDev opened the pull request:
Currently smoltcp only respondes to physical layer broadcasts (255.255.255.255), this PR enables the iface to respond to subnet broadcasts, as discussed in https://tools.ietf.org/html/rfc917 chapter 8.

## Questions

- Should this check be applied to all `ip_addrs` associated with the iface?

whitequark commented:
Thanks for the PR. However, this project [does not use `cargo fmt`](https://github.com/whitequark/smoltcp/blob/master/CODE_STYLE.md).

MabezDev committed:
subnet_broadcasts:

Adds `is_local_broadcast` to the ethernet interface which checks for subnet broadcasts, which are discussed on page 8 in https://tools.ietf.org/html/rfc917. The subnet broadcast
address is derived from the first ipv4 address on the ethernet interface
only.

MabezDev committed:
Add tests for local broadcast check

MabezDev committed:
s/is_local_broadcast/is_subnet_broadcast/g
